tests/test_integration.py

```
@@ -0,0 +1,580 @@
+
+            auth.credentials_path()
+
+            # Should integrate properly
+            mock_get_dir.assert_called_once()
+
+    @pytest.mark.asyncio
+    async def test_bot_to_auth_integration(self):
+        """Test bot module integrates with auth functions."""
+        # Test homeserver discovery integration
+        mock_client = AsyncMock()
+        mock_client.discovery_info.side_effect = nio.exceptions.RemoteTransportError(
```

Ruff (0.12.2)

Undefined name `nio`

(F821)